projects/observability/src/shared/components/cartesian/d3/chart/cartesian-chart.ts

```diff
@@ -348,6 +368,14 @@ export class DefaultCartesianChart<TData> implements CartesianChart<TData> {
           this.chartContainerElement,
           this.legendPosition
         );
+        if (this.activeSeriesSubscription) {
+          this.activeSeriesSubscription.unsubscribe();
+        }
+        this.activeSeriesSubscription = this.legend.activeSeries$.subscribe(activeSeries => {
+          this.series.length = 0;
```

Comment:
Isn't series the value passed into the chart from outside? We should never be overwriting that, aren't we losing data? I'm guessing that's why we jump through hoops above to not redraw the legend, since it still has access to the old data. Instead, we should just hold the visible series in a separate variable that we update and bind that as the d3 data.

Reply:
I have used a new variable now!!